addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -801,6 +801,73 @@ def test_message_route_alias_multiple_new(self):
         new_simple = self.env['mail.test.gateway'].search([('name', '=', 'Test Subject')])
         self.assertEqual(len(new_simple), 1, 'message_process: a new mail.test should have been created')
 
+    @mute_logger('odoo.addons.mail.models.mail_thread', 'odoo.models')
+    def test_mail_catchall_domain_allowed_validation(self):
+        """ Check the validation of `mail.catchall.domain.allowed` system parameter"""
+        for value in (',', ',,'):
+            with self.assertRaises(exceptions.ValidationError):
+                self.env['ir.config_parameter'].set_param('mail.catchall.domain.allowed', value)
+
+        for value in ('', 'hello.com', 'hello.com,,', 'hello.com,bonjour.com', 'hello.com, bonjour.com'):
+            self.env['ir.config_parameter'].set_param('mail.catchall.domain.allowed', value)
+            value = ",".join([domain.strip() for domain in value.split(',') if domain]) if value else False
```

Comment:
Note that it would probably be better to give expected value, otherwise you might have issues in this line that hide potential errors. Idea is generally to have something explicit like
```
for input, expected in [
    ('', ''),
    ('hello.com', 'hello.com'),
    ('hello.com, bonjour.com', 'hello.com,bonjour.com'),
    ('hello.COM, BONJOUR.com', 'hello.com,bonjour.com'),
]:
    ...
    self.assetEqual(parsed_input, expected)
```